### ISSUE TITLE
Corrupted completion while one candidate is prefix of another

### ISSUE THREAD

**Issue by blahgeek** [jonathanslenders/python-prompt-toolkit#384]
try this code:

```
from prompt_toolkit.contrib.completers import WordCompleter
from prompt_toolkit import prompt

animal_completer = WordCompleter([ 'cat', 'cat_and_dog', ])

if __name__ == '__main__':
    text = prompt('Give some animals: ', completer=animal_completer,
                  complete_while_typing=False)
    print('You said: %s' % text)
```

Type `c`, then two `<TAB>`, the result is `catat`, not expected `cat`.

**Comment by jonathanslenders** [comment]
Hi @blahgeek,

This is probably the issue that was fixed in the following commit: https://github.com/jonathanslenders/python-prompt-toolkit/commit/970c763323749f044931e44fb4d39b205c7b0b85

Can you try to checkout the latest master commit and try again?

**Comment by blahgeek** [comment]
no, it's the same :( @jonathanslenders

**Comment by blahgeek** [comment]
Sorry, I was wrong, it was indeed fixed in 970c763. I'm going to close this issue. Thanks for your reply.

**Comment by jonathanslenders** [comment]
Don't worry. Thanks for reporting!
